Sources/SwiftProtobuf/AnyMessageStorage.swift

```diff
@@ -259,27 +266,35 @@ extension AnyMessageStorage {
       return false
     }
 
-    // If we have both data's, compare those.
-    // The best option is to decode and compare the messages; this
-    // insulates us from variations in serialization details.  For
-    // example, one Any might hold protobuf binary bytes from one
-    // language implementation and the other from another language
-    // implementation.  But of course this only works if we
-    // actually know the message type.
-    //if let myMessage = _message {
-    //    if let otherMessage = other._message {
-    //        ... compare them directly
-    //    } else {
-    //        ... try to decode other and compare
-    //    }
-    //} else if let otherMessage = other._message {
-    //    ... try to decode ourselves and compare
-    //} else {
-    //    ... try to decode both and compare
-    //}
-    // If we don't know the message type, we have few options:
-    // If we were both deserialized from proto, compare the binary value:
-    // If we were both deserialized from JSON, compare content of the JSON?
+    // Since the library does lazy Any decode, equality is a very hard problem.
+    // It things exactly match, that's pretty easy, otherwise, one ends up having
+    // to error on saying they aren't equal.
+    //
+    // The best option would be to have Message forms and compare those, as that
+    // removes issues like map<> serialization order, some other protocol buffer
+    // implementation details/bugs around serialized form order, etc.; but that
+    // would also greatly slow down equality tests.
+    //
+    // Despite the problems with binary serialization, that may be the best bet
+    // as looking at most other language implementations around Any they don't
+    // do lazy decode and instead keep all the `value` fields in binary form and
+    // end up just comparing those binary fields when comparing Anys.
+
+    // If both have messages, check if they are the same.
```

Comment:
Yea, adding something to Message was all I could come up with.  Added this in the new commits.